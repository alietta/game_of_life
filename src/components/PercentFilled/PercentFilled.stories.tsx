import React from 'react';
import { DarkMode, LightMode, Layer } from 'sancho';
import { PercentFilled } from './index';

export default { title: 'inputs' };

export const Percent = () => (
  <div css={{ width: 400 }}>
    <DarkMode>
      <Layer css={{ borderRadius: 0 }}>
        <PercentFilled />
      </Layer>
    </DarkMode>
    <LightMode>
      <Layer css={{ borderRadius: 0 }}>
        <PercentFilled />
      </Layer>
    </LightMode>
  </div>
);
