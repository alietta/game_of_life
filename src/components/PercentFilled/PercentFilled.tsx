import React, { FC, useState, ChangeEvent } from 'react';
import { InputGroup, Input } from 'sancho';

export interface PercentFilledProps {
  onValueChange?: (value: number) => void;
}

const PercentFilled: FC<PercentFilledProps> = props => {
  const { onValueChange } = props;
  const [value, setValue] = useState<number | string>(50);
  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    const { value } = event.currentTarget;
    if (value !== '') {
      const percent = Math.abs(parseInt(event.target.value, 10));
      const roundedPercent = percent <= 100 ? percent : 100;
      setValue(roundedPercent);
      onValueChange && onValueChange(roundedPercent);
    } else {
      setValue(value);
      onValueChange && onValueChange(0);
    }
  };

  return (
    <InputGroup
      label="Percent filled"
      css={{ display: 'flex', flexDirection: 'column' }}
    >
      <Input
        inputSize="lg"
        type="number"
        placeholder="0"
        value={value}
        onChange={handleChange}
      />
    </InputGroup>
  );
};

export { PercentFilled };
