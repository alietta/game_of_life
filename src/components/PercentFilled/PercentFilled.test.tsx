import React from 'react';
import { mount } from 'enzyme';
import { PercentFilled } from './PercentFilled';

describe('PercentFilled', () => {
  it('test render', () => {
    const element = mount(<PercentFilled />);
    expect(element).toMatchSnapshot();
  });
  it('should call onValueChange with props', () => {
    const onValueChange = jest.fn();
    const element = mount(<PercentFilled onValueChange={onValueChange} />);
    element.find('input').simulate('change', {
      target: { value: 34 },
    });
    expect(onValueChange).toHaveBeenCalledWith(34);
  });
  it('should call onValueChange with wrong percent more than 100', () => {
    const onValueChange = jest.fn();
    const element = mount(<PercentFilled onValueChange={onValueChange} />);
    element.find('input').simulate('change', {
      target: { value: 315 },
    });
    expect(onValueChange).toHaveBeenCalledWith(100);
  });
  it('should call onValueChange with wrong percent less than 0', () => {
    const onValueChange = jest.fn();
    const element = mount(<PercentFilled onValueChange={onValueChange} />);
    element.find('input').simulate('change', {
      target: { value: -4 },
    });
    expect(onValueChange).toHaveBeenCalledWith(4);
  });
});
