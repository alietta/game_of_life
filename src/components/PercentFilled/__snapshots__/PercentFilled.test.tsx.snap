// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PercentFilled test render 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-3 {
  margin-top: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-3:first-child {
  margin-top: 0;
}

.emotion-1 {
  display: inline-block;
  margin-bottom: 0.5rem;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  font-weight: 400;
  line-height: 1.5;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
  font-size: 1rem;
  color: #212529;
  -webkit-font-smoothing: antialiased;
  -webkit-text-size-adjust: none;
  font-size: 0.875rem;
  font-weight: 400;
  color: #5F6871;
  line-height: 1.5;
}

.emotion-2 {
  display: block;
  width: 100%;
  line-height: 1.5;
  color: #212529;
  background-color: transparent;
  background-image: none;
  background-clip: padding-box;
  -webkit-font-smoothing: antialiased;
  -webkit-tap-highlight-color: transparent;
  -webkit-appearance: none;
  box-sizing: border-box;
  touch-action: manipulation;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
  border: none;
  box-shadow: 0 0 0 2px transparent inset,0 0 0 1px hsla(210,10.8%,14.5%,0.2) inset;
  border-radius: 0.25rem;
  -webkit-transition: background 0.25s cubic-bezier(0.35,0,0.25,1),border-color 0.15s cubic-bezier(0.35,0,0.25,1),box-shadow 0.15s cubic-bezier(0.35,0,0.25,1);
  transition: background 0.25s cubic-bezier(0.35,0,0.25,1),border-color 0.15s cubic-bezier(0.35,0,0.25,1),box-shadow 0.15s cubic-bezier(0.35,0,0.25,1);
  font-size: 1.25rem;
  padding: 0.65rem 1rem;
  height: 48px;
}

.emotion-2::-webkit-input-placeholder {
  color: hsla(210,10.8%,14.5%,0.45);
}

.emotion-2::-moz-placeholder {
  color: hsla(210,10.8%,14.5%,0.45);
}

.emotion-2:-ms-input-placeholder {
  color: hsla(210,10.8%,14.5%,0.45);
}

.emotion-2::placeholder {
  color: hsla(210,10.8%,14.5%,0.45);
}

.emotion-2:focus {
  box-shadow: 0 0 0 3px hsla(209,75.4%,38.2%,0.1),inset 0 0 0 1px hsla(210,10.8%,14.5%,0.2),inset 0 -1px 1px 0 hsla(210,10.8%,14.5%,0.05);
  outline: none;
}

.emotion-2:disabled {
  opacity: 0.8;
  background: #f1f3f5;
  cursor: not-allowed;
  box-shadow: 0 0 0 2px transparent inset,0 0 0 1px hsla(210,10.8%,14.5%,0.12) inset;
}

.emotion-2:active {
  background: #f1f3f5;
}

<PercentFilled>
  <Component
    className="emotion-4"
    label="Percent filled"
  >
    <div
      className="emotion-3"
    >
      <Component
        htmlFor="1"
      >
        <label
          className="Label emotion-1"
          htmlFor="1"
        >
          <Component
            className="Label__text"
            variant="subtitle"
          >
            <div
              className="Label__text emotion-0"
            >
              Percent filled
            </div>
          </Component>
        </label>
      </Component>
      <ForwardRef
        inputSize="lg"
        onChange={[Function]}
        placeholder="0"
        type="number"
        value={50}
      >
        <input
          className="Input emotion-2"
          id="1"
          onChange={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          placeholder="0"
          type="number"
          value={50}
        />
      </ForwardRef>
    </div>
  </Component>
</PercentFilled>
`;
